### ISSUE TITLE
Investigate why lots of tests are failing on Windows

### ISSUE THREAD

**Issue by ahayzen-kdab** [KDAB/cxx-qt#111]
Currently these tests are failing on Windows `cxx_qt_gen_cargo_tests|example_qml.*|qml.*tests|test.*|cpp_clang_format` some of these are related for formatted and some are not building correctly. We need to setup a build env on Windows to check what is happening as no one has tried it yet :-)

**Comment by OlivierLDff** [comment]
About valgrind tests, wouldn't it be easier to add a check if valgrind has been found. Instead of having to exclude it manually. Better not generate them? A warning could be emitted on non windows system.
https://github.com/KDAB/cxx-qt/blob/c435d65aba8eef451596def5f29c3b52dae8ea23/CMakeLists.txt#L134-L139

**Comment by ahayzen-kdab** [comment]
Could do, i was also wondering eventually whether ctest is the best route with github actions anyway, and instead splitting things into separate tasks might be better. So then valgrind could be a github action task that is only run on Linux, same with reuse, clang-format etc. Then these can be run in parallel to potentially speed up CI (another project we have does this), but it's low priority for probably not much gain.

So having something with ignores the test when valgrind isn't around is probably best for now.

**Comment by OlivierLDff** [comment]
I would argue than less code there is in github action, the easier it is to reproduce ci environment locally and when things don't work. So being able to just trigger ctest is nice and select the test to run.
That being said, you are right that task like reuse, clang format, could be removed from there. But I don't think there is much to gain in CI speed, as compilation take the most of the time.

**Comment by ahayzen-kdab** [comment]
Yup, having a way to run them locally is also why ctest still exists as the route :-) Just adding them all in there is fine for now.
